src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/security/JwtService.kt

```diff
@@ -41,25 +39,24 @@ class JwtService(jwtConfig: JwtConfig) {
     Jwts.builder()
       .setId(UUID.randomUUID().toString())
       .setSubject(email)
-      .setExpiration(
-        Date.from(
-          LocalDateTime.now().plus(expiry.toMillis(), ChronoUnit.MILLIS).toInstant(zoneOffset())
-        )
-      )
+      .setExpiration(Date.from(Instant.now().plus(expiry.toMillis(), ChronoUnit.MILLIS)))
+      .addClaims(mapOf("authorities" to arrayOf("ROLE_SLM_CREATE_BARCODE")))
       .signWith(SignatureAlgorithm.RS256, privateKey)
       .compact()
 
-  private fun zoneOffset() = ZoneId.systemDefault().rules.getOffset(Instant.now())
-
   fun validateToken(jwt: String): Boolean =
     runCatching {
       Jwts.parser().setSigningKey(publicKey).parseClaimsJws(jwt)
     }.onFailure {
       if ((it is ExpiredJwtException).not()) {
-        log.warn("Found an invalid JWT: jwt", it)
+        log.warn("Found an invalid JWT: $jwt", it)
       }
     }.isSuccess
 
   fun subject(jwt: String): String =
     Jwts.parser().setSigningKey(publicKey).parseClaimsJws(jwt).body.subject
+
+  @Suppress("UNCHECKED_CAST")
+  fun authorities(jwt: String): List<String>? =
+    Jwts.parser().setSigningKey(publicKey).parseClaimsJws(jwt).body["authorities"] as? List<String>
```

Comment:
The type of body is a Map<String, Object> but we know that if the authorities are there then they will be a list